packages/capabilities/src/types.ts

```diff
@@ -449,6 +452,37 @@ export type UploadGetFailure = UploadNotFound | Ucanto.Failure
 // HTTP
 export type HTTPPut = InferInvokedCapability<typeof HTTPCaps.put>
 
+// Index
+export type Index = InferInvokedCapability<typeof IndexCaps.index>
+export type IndexAdd = InferInvokedCapability<typeof IndexCaps.add>
+
+export type IndexAddSuccess = Unit
+
+export type IndexAddFailure =
+  | UnknownFormat
+  | ShardNotFound
+  | SliceNotFound
+  | Failure
+
+/** The index is not in a format understood by the service. */
+export interface UnknownFormat extends Failure {
+  name: 'UnknownFormat'
+}
+
+/** A shard referenced by the index is not stored in the referenced space. */
+export interface ShardNotFound extends Failure {
+  name: 'ShardNotFound'
+  /** Multihash digest of the shard that could not be found. */
+  digest: Multihash
+}
+
+/** A slice referenced by the index was not found in the specified shard. */
+export interface SliceNotFound extends Failure {
+  name: 'SliceNotFound'
+  /** Multihash digest of the slice that could not be found. */
+  digest: Multihash
+}
```

Comment:
just a question, are we going to validate this?

Reply:
We have talked previously about sampling a few in https://github.com/w3s-project/specs/pull/85